samza-core/src/main/java/org/apache/samza/container/RunLoop.java

```diff
@@ -185,6 +180,11 @@ public void run() {
           containerMetrics.utilization().set(((double) activeNs) / totalNs);
         }
       }
+
+      if (throwable != null) {
```

Comment:
Is the following scenario possible?
1) Async processing has a bad message in flight
2) Manual call to `RunLoop.shutdown`
3) Exit main while loop
4) `throwable` is still null since async processing is incomplete
5) Async processing throws exception
6) Container will exit with "success"

Reply:
That is correct. The semantics of manual call to `RunLoop.shutdown` is loosely defined in the sense that we don't wait for things in flight to finish and trigger shutdown as soon as runloop thread sees the request. So this would be in line with that and may or may not propagate the exception.

Reply:
Could you please add a note somewhere (maybe javadoc and/or PR description) about this scenario? The PR title/description seems to refer to the general shutdown case (which would include `RunLoop.shutdown`), but it looks like you are focusing on the end-of-stream shutdown case in this PR.

Reply:
Added a java doc, updated the PR description and also added some details to the JIRA ticket to outline what is this PR targeting.